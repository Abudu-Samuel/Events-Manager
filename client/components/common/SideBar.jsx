import React from 'react';
import { Link } from 'react-router-dom';


const SideBar = () => (
  <div className="col-md-4 adm mb-4">
    <div className="col-xl-5 col-md-4 mb-3 try text-center">
      <img src="http://res.cloudinary.com/leumas/image/upload/v1522632122/mn5bz05wvv7qfkozmzmc.png" className="img-fluid mb-3 teal-text z-depth-1 rounded-circle admpro" alt="Responsive image" />
      <h5 className="font-weight-bold white-text">Admin</h5>
    </div>
    {/* <h6 className="font-weight-bold white-text text-center">adminuser@gmail.com</h6> */}
    <hr className="font-weight-bold" />
    <div className="">
      <ul className="mr-6">
        <li className="white-text mb-3">
          <a className="btn btn-mycolor btn-block mr-2" href="#"><i className="fa fa-dashboard mr-2 white-text" />Dashboard</a>
        </li>
        {/* <li className="white-text mb-3">
          <a className="btn btn-mycolor btn-block mr-4" href="#"><i className="fa fa-user mr-2 white-text" />Profile</a>
        </li> */}
        <li className="white-text mb-3">
          <Link className="btn btn-mycolor btn-block mr-4" to="/addcenter"><i className="fa fa-plus mr-2 white-text" />Add Center</Link>
        </li>
        <li className="white-text mb-3">
          <Link to="/manage/center" className="btn btn-mycolor btn-block mr-4"><i className="fa fa-street-view mr-2 white-text" />View Centers</Link>
        </li>
        {/* <li className="white-text mb-3">
          <a className="btn btn-mycolor btn-block mr-4" href="#"><i className="fa fa-cog mr-2 white-text" />Settings</a>
        </li> */}
      </ul>
    </div>
  </div>
);

export default SideBar;
