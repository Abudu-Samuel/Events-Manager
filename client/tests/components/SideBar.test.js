import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import SideBar from '../../components/common/SideBar';

describe('<SideBar />', () => {
  it('should render SideBar correctly', () => {
    const wrapper = shallow(<SideBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
