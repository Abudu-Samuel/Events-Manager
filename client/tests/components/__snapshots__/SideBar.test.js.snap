// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> should render SideBar correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-xl-5 col-md-4 mb-3 try text-center"
        >
          <img
            className="img-fluid mb-3 teal-text z-depth-1 rounded-circle admpro"
            src="https://res.cloudinary.com/leumas/image/upload/v1527177652/wnmjpssnmvcewty3huqf.png"
          />
          <h5
            className="font-weight-bold min white-text"
          >
            Admin
          </h5>
        </div>,
        <hr
          className="font-weight-bold"
        />,
        <div
          className=""
        >
          <ul
            className="mr-6"
            include="hello"
          >
            <li
              className="white-text mb-3"
            >
              <Link
                className="btn btn-mycolor btn-block mr-2"
                replace={false}
                to="/admin/dashboard"
              >
                <i
                  className="fa fa-dashboard mr-2 white-text"
                />
                Dashboard
              </Link>
            </li>
            <li
              className="white-text mb-3"
            >
              <Link
                className="btn btn-mycolor ad btn-block mr-4"
                replace={false}
                to="/addcenter"
              >
                <i
                  className="fa fa-plus mr-2 white-text"
                />
                Add Center
              </Link>
            </li>
            <li
              className="white-text mb-3"
            >
              <Link
                className="btn vc btn-mycolor btn-block mr-4"
                replace={false}
                to="/manage/centers"
              >
                <i
                  className="fa fa-street-view mr-2 white-text"
                />
                View Centers
              </Link>
            </li>
          </ul>
        </div>,
      ],
      "className": "col-md-4 adm mb-4",
      "style": Object {
        "marginTop": 19,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="img-fluid mb-3 teal-text z-depth-1 rounded-circle admpro"
              src="https://res.cloudinary.com/leumas/image/upload/v1527177652/wnmjpssnmvcewty3huqf.png"
            />,
            <h5
              className="font-weight-bold min white-text"
            >
              Admin
            </h5>,
          ],
          "className": "col-xl-5 col-md-4 mb-3 try text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "img-fluid mb-3 teal-text z-depth-1 rounded-circle admpro",
              "src": "https://res.cloudinary.com/leumas/image/upload/v1527177652/wnmjpssnmvcewty3huqf.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Admin",
              "className": "font-weight-bold min white-text",
            },
            "ref": null,
            "rendered": "Admin",
            "type": "h5",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "font-weight-bold",
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="mr-6"
            include="hello"
          >
            <li
              className="white-text mb-3"
            >
              <Link
                className="btn btn-mycolor btn-block mr-2"
                replace={false}
                to="/admin/dashboard"
              >
                <i
                  className="fa fa-dashboard mr-2 white-text"
                />
                Dashboard
              </Link>
            </li>
            <li
              className="white-text mb-3"
            >
              <Link
                className="btn btn-mycolor ad btn-block mr-4"
                replace={false}
                to="/addcenter"
              >
                <i
                  className="fa fa-plus mr-2 white-text"
                />
                Add Center
              </Link>
            </li>
            <li
              className="white-text mb-3"
            >
              <Link
                className="btn vc btn-mycolor btn-block mr-4"
                replace={false}
                to="/manage/centers"
              >
                <i
                  className="fa fa-street-view mr-2 white-text"
                />
                View Centers
              </Link>
            </li>
          </ul>,
          "className": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="white-text mb-3"
              >
                <Link
                  className="btn btn-mycolor btn-block mr-2"
                  replace={false}
                  to="/admin/dashboard"
                >
                  <i
                    className="fa fa-dashboard mr-2 white-text"
                  />
                  Dashboard
                </Link>
              </li>,
              <li
                className="white-text mb-3"
              >
                <Link
                  className="btn btn-mycolor ad btn-block mr-4"
                  replace={false}
                  to="/addcenter"
                >
                  <i
                    className="fa fa-plus mr-2 white-text"
                  />
                  Add Center
                </Link>
              </li>,
              <li
                className="white-text mb-3"
              >
                <Link
                  className="btn vc btn-mycolor btn-block mr-4"
                  replace={false}
                  to="/manage/centers"
                >
                  <i
                    className="fa fa-street-view mr-2 white-text"
                  />
                  View Centers
                </Link>
              </li>,
            ],
            "className": "mr-6",
            "include": "hello",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="btn btn-mycolor btn-block mr-2"
                  replace={false}
                  to="/admin/dashboard"
                >
                  <i
                    className="fa fa-dashboard mr-2 white-text"
                  />
                  Dashboard
                </Link>,
                "className": "white-text mb-3",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <i
                      className="fa fa-dashboard mr-2 white-text"
                    />,
                    "Dashboard",
                  ],
                  "className": "btn btn-mycolor btn-block mr-2",
                  "replace": false,
                  "to": "/admin/dashboard",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-dashboard mr-2 white-text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "Dashboard",
                ],
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="btn btn-mycolor ad btn-block mr-4"
                  replace={false}
                  to="/addcenter"
                >
                  <i
                    className="fa fa-plus mr-2 white-text"
                  />
                  Add Center
                </Link>,
                "className": "white-text mb-3",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <i
                      className="fa fa-plus mr-2 white-text"
                    />,
                    "Add Center",
                  ],
                  "className": "btn btn-mycolor ad btn-block mr-4",
                  "replace": false,
                  "to": "/addcenter",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-plus mr-2 white-text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "Add Center",
                ],
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="btn vc btn-mycolor btn-block mr-4"
                  replace={false}
                  to="/manage/centers"
                >
                  <i
                    className="fa fa-street-view mr-2 white-text"
                  />
                  View Centers
                </Link>,
                "className": "white-text mb-3",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <i
                      className="fa fa-street-view mr-2 white-text"
                    />,
                    "View Centers",
                  ],
                  "className": "btn vc btn-mycolor btn-block mr-4",
                  "replace": false,
                  "to": "/manage/centers",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-street-view mr-2 white-text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "View Centers",
                ],
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-xl-5 col-md-4 mb-3 try text-center"
          >
            <img
              className="img-fluid mb-3 teal-text z-depth-1 rounded-circle admpro"
              src="https://res.cloudinary.com/leumas/image/upload/v1527177652/wnmjpssnmvcewty3huqf.png"
            />
            <h5
              className="font-weight-bold min white-text"
            >
              Admin
            </h5>
          </div>,
          <hr
            className="font-weight-bold"
          />,
          <div
            className=""
          >
            <ul
              className="mr-6"
              include="hello"
            >
              <li
                className="white-text mb-3"
              >
                <Link
                  className="btn btn-mycolor btn-block mr-2"
                  replace={false}
                  to="/admin/dashboard"
                >
                  <i
                    className="fa fa-dashboard mr-2 white-text"
                  />
                  Dashboard
                </Link>
              </li>
              <li
                className="white-text mb-3"
              >
                <Link
                  className="btn btn-mycolor ad btn-block mr-4"
                  replace={false}
                  to="/addcenter"
                >
                  <i
                    className="fa fa-plus mr-2 white-text"
                  />
                  Add Center
                </Link>
              </li>
              <li
                className="white-text mb-3"
              >
                <Link
                  className="btn vc btn-mycolor btn-block mr-4"
                  replace={false}
                  to="/manage/centers"
                >
                  <i
                    className="fa fa-street-view mr-2 white-text"
                  />
                  View Centers
                </Link>
              </li>
            </ul>
          </div>,
        ],
        "className": "col-md-4 adm mb-4",
        "style": Object {
          "marginTop": 19,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="img-fluid mb-3 teal-text z-depth-1 rounded-circle admpro"
                src="https://res.cloudinary.com/leumas/image/upload/v1527177652/wnmjpssnmvcewty3huqf.png"
              />,
              <h5
                className="font-weight-bold min white-text"
              >
                Admin
              </h5>,
            ],
            "className": "col-xl-5 col-md-4 mb-3 try text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "img-fluid mb-3 teal-text z-depth-1 rounded-circle admpro",
                "src": "https://res.cloudinary.com/leumas/image/upload/v1527177652/wnmjpssnmvcewty3huqf.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Admin",
                "className": "font-weight-bold min white-text",
              },
              "ref": null,
              "rendered": "Admin",
              "type": "h5",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "font-weight-bold",
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="mr-6"
              include="hello"
            >
              <li
                className="white-text mb-3"
              >
                <Link
                  className="btn btn-mycolor btn-block mr-2"
                  replace={false}
                  to="/admin/dashboard"
                >
                  <i
                    className="fa fa-dashboard mr-2 white-text"
                  />
                  Dashboard
                </Link>
              </li>
              <li
                className="white-text mb-3"
              >
                <Link
                  className="btn btn-mycolor ad btn-block mr-4"
                  replace={false}
                  to="/addcenter"
                >
                  <i
                    className="fa fa-plus mr-2 white-text"
                  />
                  Add Center
                </Link>
              </li>
              <li
                className="white-text mb-3"
              >
                <Link
                  className="btn vc btn-mycolor btn-block mr-4"
                  replace={false}
                  to="/manage/centers"
                >
                  <i
                    className="fa fa-street-view mr-2 white-text"
                  />
                  View Centers
                </Link>
              </li>
            </ul>,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="white-text mb-3"
                >
                  <Link
                    className="btn btn-mycolor btn-block mr-2"
                    replace={false}
                    to="/admin/dashboard"
                  >
                    <i
                      className="fa fa-dashboard mr-2 white-text"
                    />
                    Dashboard
                  </Link>
                </li>,
                <li
                  className="white-text mb-3"
                >
                  <Link
                    className="btn btn-mycolor ad btn-block mr-4"
                    replace={false}
                    to="/addcenter"
                  >
                    <i
                      className="fa fa-plus mr-2 white-text"
                    />
                    Add Center
                  </Link>
                </li>,
                <li
                  className="white-text mb-3"
                >
                  <Link
                    className="btn vc btn-mycolor btn-block mr-4"
                    replace={false}
                    to="/manage/centers"
                  >
                    <i
                      className="fa fa-street-view mr-2 white-text"
                    />
                    View Centers
                  </Link>
                </li>,
              ],
              "className": "mr-6",
              "include": "hello",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="btn btn-mycolor btn-block mr-2"
                    replace={false}
                    to="/admin/dashboard"
                  >
                    <i
                      className="fa fa-dashboard mr-2 white-text"
                    />
                    Dashboard
                  </Link>,
                  "className": "white-text mb-3",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fa fa-dashboard mr-2 white-text"
                      />,
                      "Dashboard",
                    ],
                    "className": "btn btn-mycolor btn-block mr-2",
                    "replace": false,
                    "to": "/admin/dashboard",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fa fa-dashboard mr-2 white-text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    "Dashboard",
                  ],
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="btn btn-mycolor ad btn-block mr-4"
                    replace={false}
                    to="/addcenter"
                  >
                    <i
                      className="fa fa-plus mr-2 white-text"
                    />
                    Add Center
                  </Link>,
                  "className": "white-text mb-3",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fa fa-plus mr-2 white-text"
                      />,
                      "Add Center",
                    ],
                    "className": "btn btn-mycolor ad btn-block mr-4",
                    "replace": false,
                    "to": "/addcenter",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fa fa-plus mr-2 white-text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    "Add Center",
                  ],
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="btn vc btn-mycolor btn-block mr-4"
                    replace={false}
                    to="/manage/centers"
                  >
                    <i
                      className="fa fa-street-view mr-2 white-text"
                    />
                    View Centers
                  </Link>,
                  "className": "white-text mb-3",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fa fa-street-view mr-2 white-text"
                      />,
                      "View Centers",
                    ],
                    "className": "btn vc btn-mycolor btn-block mr-4",
                    "replace": false,
                    "to": "/manage/centers",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fa fa-street-view mr-2 white-text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    "View Centers",
                  ],
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
